Source/DataSources/EntityCollection.js

```diff
@@ -30,15 +30,27 @@ define([
     };
 
     function fireChangedEvent(collection) {
+        if (collection._firing) {
```

Comment:
Whoops.  What's weird is the test I wrote fails in master and passes in this branch (doing the right thing as far as I can tell).  But what I'm not catching is checking that the second event is not fired until the first even is finished, which might be hard to test.  I'll look into it.